test/widgets/action_sheet_test.dart

```diff
@@ -99,6 +103,302 @@ void main() {
+                                            label: 'Follow topic');
+    final unfollow =             findButton(from:  UserTopicVisibilityPolicy.followed,
+                                            to:    UserTopicVisibilityPolicy.none,
+                                            label: 'Unfollow topic');
+
+    /// Prepare store and bring up a topic action sheet.
+    ///
+    /// If `isChannelMuted` is `null`, the user is not subscribed to the
+    /// channel.
+    Future<void> setupToTopicActionSheet(WidgetTester tester, {
+      required bool? isChannelMuted,
+      required UserTopicVisibilityPolicy visibilityPolicy,
+      int? featureLevel,
+    }) async {
+      addTearDown(testBinding.reset);
+
+      channel = eg.stream();
+      topic = 'isChannelMuted: $isChannelMuted, policy: $visibilityPolicy';
+
+      final account = eg.selfAccount.copyWith(zulipFeatureLevel: featureLevel);
+      final subscriptions = isChannelMuted == null ? <Subscription>[]
+        : [eg.subscription(channel, isMuted: isChannelMuted)];
+      await testBinding.globalStore.add(account, eg.initialSnapshot(
+        realmUsers: [eg.selfUser],
+        streams: [channel],
+        subscriptions: subscriptions,
+        userTopics: [eg.userTopicItem(channel, topic, visibilityPolicy)],
+        zulipFeatureLevel: featureLevel));
+      store = await testBinding.globalStore.perAccount(account.id);
+      connection = store.connection as FakeApiConnection;
+
+      connection.prepare(json: eg.newestGetMessagesResult(
+        foundOldest: true, messages: [
+          eg.streamMessage(stream: channel, topic: topic)]).toJson());
+      await tester.pumpWidget(TestZulipApp(accountId: account.id,
+        child: MessageListPage(
+          initNarrow: TopicNarrow(channel.streamId, topic))));
+      await tester.pumpAndSettle();
+
+      await tester.longPress(find.descendant(
+        of: find.byType(RecipientHeader), matching: find.text(topic)));
+      // sheet appears onscreen; default duration of bottom-sheet enter animation
+      await tester.pump(const Duration(milliseconds: 250));
+    }
+
+    void checkButtons(List<Finder> expectedButtonFinders) {
+      if (expectedButtonFinders.isEmpty) {
+        check(find.byType(BottomSheet)).findsNothing();
+        return;
+      }
+      check(find.byType(BottomSheet)).findsOne();
+
+      for (final buttonFinder in expectedButtonFinders) {
+        check(buttonFinder).findsOne();
+      }
+      check(find.bySubtype<UserTopicUpdateButton>())
+        .findsExactly(expectedButtonFinders.length);
+    }
+
+    void checkUpdateUserTopicRequest(UserTopicVisibilityPolicy expectedPolicy) async {
+      check(connection.lastRequest).isA<http.Request>()
+        ..url.path.equals('/api/v1/user_topics')
+        ..bodyFields.deepEquals({
+          'stream_id': '${channel.streamId}',
+          'topic': topic,
+          'visibility_policy': '${expectedPolicy.toJson()}',
```

Comment:
```suggestion
          'visibility_policy': jsonConvert(expectedPolicy),
```

Just calling `toString` (or doing string interpolation, which calls `toString`) on the `toJson` result happens to work here, because the toJson result is an int. But it doesn't get a JSON encoding in general. Better to just use the always-correct thing, which isn't any more complicated to write.